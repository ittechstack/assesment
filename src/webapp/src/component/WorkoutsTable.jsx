import React, {useState} from 'react';
import 'react-toastify/dist/ReactToastify.css';
import {AddWorkoutModal} from "./AddWorkoutModal";

export const WorkoutsTable = props => {

    const {trainer, refreshWorkoutsOfTrainer} = props;

    const [showAddWorkout, setShowAddWorkout] = useState(false);

    return (
        <div className="col-sm">
            <AddWorkoutModal trainer={trainer}
                             show={showAddWorkout}
                             onHide={() => {
                                 setShowAddWorkout(false);
                                 refreshWorkoutsOfTrainer(trainer);
                             }}
            />

            {
                trainer &&
                <>
                    <h3>Workouts of <span style={{color: "red"}}>{trainer.name}</span></h3>
                    <table className="table">
                        <thead>
                        <tr>
                            <th>Workout Name</th>
                        </tr>
                        </thead>
                        <tbody>
                        {
                            trainer.workouts.map(workout => {
                                return <tr key={workout.id}>
                                    <td>{workout.name}</td>
                                </tr>
                            })
                        }
                        </tbody>
                    </table>
                    <div className="d-flex flex-row-reverse">
                        <button className="btn btn-success" onClick={() => setShowAddWorkout(true)}>Add Workout</button>
                    </div>
                </>
            }
        </div>
    )
};