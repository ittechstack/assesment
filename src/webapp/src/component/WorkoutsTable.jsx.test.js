import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer';
import {WorkoutsTable} from "./WorkoutsTable";

configure({adapter: new Adapter()});

describe('WorkoutsTable component testing', () => {
    it('render correctly WorkoutsTable component', () => {
        const trainer = {
            id: 1,
            name: 'Tom',
            workouts: [{
                id: 1,
                name: 'wo1'
            }]
        };
        const wrapper = renderer.create(<WorkoutsTable trainer={trainer}/>).toJSON();

        expect(wrapper).toMatchSnapshot();
    });

    it('should render header', () => {
        const trainer = {
            id: 1,
            name: 'Tom',
            workouts: [{
                id: 1,
                name: 'wo1'
            }]
        };

        const wrapper = shallow(<WorkoutsTable trainer={trainer}/>);

        const header = <h3>Workouts of <span style={{color: "red"}}>{trainer.name}</span></h3>;

        expect(wrapper.contains(header)).toBe(true)
    });

    it('should render workouts table with 2 records', () => {
        const trainer = {
            id: 1,
            name: 'Tom',
            workouts: [
                {
                    id: 1,
                    name: 'wo1'
                },
                {
                    id: 2,
                    name: 'wo2'
                }
            ]
        };

        const wrapper = shallow(<WorkoutsTable trainer={trainer}/>);

        const table = wrapper.find('table');
        expect(table).toHaveLength(1);

        const tbody = table.find('tbody');
        expect(tbody).toHaveLength(1);

        const rows = tbody.find('tr');
        expect(rows).toHaveLength(trainer.workouts.length);
    });
});