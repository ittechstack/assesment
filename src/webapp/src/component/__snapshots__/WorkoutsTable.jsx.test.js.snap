// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkoutsTable component testing render correctly WorkoutsTable component 1`] = `
<div
  className="col-sm"
>
  <h3>
    Workouts of 
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
      Tom
    </span>
  </h3>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          Workout Name
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          wo1
        </td>
      </tr>
    </tbody>
  </table>
  <div
    className="d-flex flex-row-reverse"
  >
    <button
      className="btn btn-success"
      onClick={[Function]}
    >
      Add Workout
    </button>
  </div>
</div>
`;
